backend/FwLite/LocalWebApp/Services/ImportFwdataService.cs

```diff
@@ -10,23 +10,33 @@ public class ImportFwdataService(ProjectsService projectsService, ILogger<Import
 {
     public async Task<CrdtProject> Import(string projectName)
     {
+        var startTime = Stopwatch.GetTimestamp();
         var fwDataProject = FieldWorksProjectList.GetProject(projectName);
         if (fwDataProject is null)
         {
             throw new InvalidOperationException($"Project {projectName} not found.");
         }
         using var fwDataApi = fwDataFactory.GetFwDataMiniLcmApi(fwDataProject, false);
-        var project = await projectsService.CreateProject(fwDataProject.Name,
-            afterCreate: async (provider, project) =>
-            {
-                var crdtApi = provider.GetRequiredService<ILexboxApi>();
-                await ImportProject(crdtApi, fwDataApi, fwDataApi.EntryCount);
-            });
-        logger.LogInformation("Import of {ProjectName} complete!", fwDataApi.Project.Name);
-        return project;
+        try
+        {
+            var project = await projectsService.CreateProject(fwDataProject.Name,
+                afterCreate: async (provider, project) =>
+                {
+                    var crdtApi = provider.GetRequiredService<ILexboxApi>();
+                    await ImportProject(crdtApi, fwDataApi, fwDataApi.EntryCount);
+                });
+            var timeSpent = Stopwatch.GetElapsedTime(startTime);
+            logger.LogInformation("Import of {ProjectName} complete, took {TimeSpend}", fwDataApi.Project.Name, timeSpent.Humanize(2));
+            return project;
+        }
+        catch
+        {
+            logger.LogError("Import of {ProjectName} failed, deleting project", fwDataApi.Project.Name);
```

Comment:
Would it be useful to log the exception too?

Reply:
since we're throwing the exception below I decided not to, otherwise it may get logged multiple times which would be confusing. For example if the code that calls this method catches and logs the exception then it would get logged twice.